src/main/haskell/kore/app/exec/Main.hs

```diff
@@ -158,6 +166,28 @@ parserInfoModifiers =
                 \in PATTERN_FILE."
     <> header "kore-exec - an interpreter for Kore definitions"
 
+externalizeFreshVars :: CommonPurePattern level -> CommonPurePattern level
+externalizeFreshVars pat = cata renameFreshLocal pat
+  where
+    allVarsIds = Set.map (getId . variableName) (pureAllVariables pat)
+    freshVarsIds = Set.filter (isPrefixOf "var_") allVarsIds
+    computeFreshPrefix :: String -> (Set.Set String) -> String
+    computeFreshPrefix pref strings
+      | Set.null matchingStrings = pref
+      | otherwise = computeFreshPrefix (pref ++ "-") matchingStrings
```

Comment:
Maybe add a comment that one may wish to move to disambiguators which are not in base 1.